webapp/tests/karma/ts/components/filters/analytics-filter.component.spec.ts

```diff
@@ -0,0 +1,136 @@
+      navigate: sinon.stub(),
+      events: of(new NavigationEnd(0, '', '')),
+    };
+
+    const mockedSelectors = [
+      { selector: Selectors.getSidebarFilter, value: {} },
+    ];
+
+    return TestBed
+      .configureTestingModule({
+        imports: [
+          TranslateModule.forRoot({ loader: { provide: TranslateLoader, useClass: TranslateFakeLoader } })
+        ],
+        declarations: [
+          AnalyticsFilterComponent,
+        ],
+        providers: [
+          provideMockStore({ selectors: mockedSelectors }),
+          { provide: AuthService, useValue: authService },
+          { provide: SessionService, useValue: sessionService },
+          { provide: TelemetryService, useValue: telemetryService },
+          { provide: UserSettingsService, useValue: userSettingsService },
+          { provide: ActivatedRoute, useValue: route },
+          { provide: Router, useValue: router },
+        ]
+      })
+      .compileComponents()
+      .then(() => {
+        fixture = TestBed.createComponent(AnalyticsFilterComponent);
+        component = fixture.componentInstance;
+        store = TestBed.inject(MockStore);
+        fixture.detectChanges();
+      });
+  }));
+
+  afterEach(() => {
+    store.resetSelectors();
+    sinon.restore();
+  });
+
+  it('should create AnalyticsFilterComponent', () => {
+    expect(component).to.exist;
+  });
+
+  it('should display filter button when all conditions of showFilterButton are true', fakeAsync(() => {
+    authService.has
+      .withArgs(['can_view_old_filter_and_search', 'can_view_old_action_bar'])
+      .resolves(false);
+    sessionService.isAdmin.returns(false);
+    userSettingsService.hasMultipleFacilities.resolves(true);
+    route.snapshot.firstChild.data.moduleId = 'target-aggregates';
+
+    component.ngOnInit();
+    flush();
+
+    expect(component.showFilterButton).to.be.true;
+  }));
+
+  it('should not display filter button if user does not have multipleFacilities', fakeAsync(() => {
+    userSettingsService.hasMultipleFacilities.resolves(false);
+
+    component.ngOnInit();
+    flush();
+
+    expect(component.showFilterButton).to.be.false;
+  }));
+
+  it('should not display filter button if user is admin', fakeAsync(() => {
+    sessionService.isAdmin.returns(true);
+
+    component.ngOnInit();
+    flush();
+
+    expect(component.showFilterButton).to.be.false;
+  }));
+
+  it('should not display filter button if user old UI permissions', fakeAsync(() => {
+    authService.has
+      .withArgs(['!can_view_old_filter_and_search', '!can_view_old_action_bar'])
+      .resolves(false);
+
+    component.ngOnInit();
+    flush();
+
+    expect(component.showFilterButton).to.be.false;
+  }));
+
+  it('should not display filter button if module is not target aggregates', fakeAsync(() => {
+    route.snapshot.firstChild.data.moduleId = 'not-target-aggregates';
+
+    component.ngOnInit();
+    flush();
+
+    expect(component.showFilterButton).to.be.false;
+  }));
```

Comment:
Please add a unit test asserting that `openSidebar() {` calls the global actions and telemetry with the right values and the right amount of time (call count of each function), when opening and closing the filter.